internal/kafka/internal/config/kafka_supported_instance_types.go

```diff
@@ -128,3 +128,21 @@ func NewKafkaSupportedInstanceTypesConfig() *KafkaSupportedInstanceTypesConfig {
 		ConfigurationFile: "config/kafka-instance-types-configuration.yaml",
 	}
 }
+
+func (s *SupportedKafkaInstanceTypesConfig) GetKafkaInstanceTypeByID(instanceType string) (KafkaInstanceType, error) {
+	for _, t := range s.SupportedKafkaInstanceTypes {
+		if t.Id == instanceType {
+			return t, nil
+		}
+	}
+	return KafkaInstanceType{}, fmt.Errorf("Unable to find kafka instance type for '%s'", instanceType)
+}
+
+func (kp *KafkaInstanceType) GetKafkaInstanceSizeByID(instanceType, sizeId string) (KafkaInstanceSize, error) {
+	for _, size := range kp.Sizes {
+		if size.Id == sizeId {
+			return size, nil
+		}
+	}
+	return KafkaInstanceSize{}, fmt.Errorf("Kafka instance size id: '%s' not found for '%s' instance type", sizeId, instanceType)
+}
```

Comment:
This would be better as:
```go
func (kp *KafkaInstanceType) GetKafkaInstanceSizeByID(instanceType, sizeId string) (*KafkaInstanceSize, error) {
	for _, size := range kp.Sizes {
		if size.Id == sizeId {
                         ret := size // copy the object to avoid exposing mutable internals
			return &ret, nil
		}
	}
	return nil, fmt.Errorf("Kafka instance size id: '%s' not found for '%s' instance type", sizeId, instanceType)
}
```

Reply:
@ziccardi what internal do you mean with `copy the object to avoid exposing mutable internals`?

Reply:
The `KafkaInstanceType` contains a list of `KafkaInstanceSize`. The list is not deeply copied: only the address is copied. That means that if the caller changes some object inside the received `KafkaInstanceType.Sizes` array, the called object gets the changes too.

That could lead to errors hard to root since the content of an internal is changed somewhere else in the code.

Reply:
@pb82 In the comment you are referring to, I mean that we need to create a copy and return the address of that copy.
If we would return `&size` directly, we would return the address of the object inside the `kp.Sizes` array, thus giving the chance to the caller to change the internal state of the object.

Reply:
However I believe that it is not really needed to make that copy here since `range` already returns a shallow copy of the object in the array

Reply:
Good spot, thanks @ziccardi